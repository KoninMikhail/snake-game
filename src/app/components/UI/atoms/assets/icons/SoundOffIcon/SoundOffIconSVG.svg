<?xml version="1.0" encoding="utf-8"?>
<!-- Generator: Adobe Illustrator 26.0.1, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
<svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
	 viewBox="0 0 24 24" style="enable-background:new 0 0 24 24;" xml:space="preserve">
<g id="Layer_1_00000059283214915903458300000000863206735705567917_">
	<path d="M11.6,2.5c0-0.1,0-0.1-0.1-0.1c-0.3,0-0.6,0-0.9,0c-0.1,0-0.1,0-0.1,0.1v0.1c0,0.3,0,0.5,0,0.8v0.1c0,0,0,0.1-0.1,0.1l0,0
		h-0.1c-0.2,0-0.5,0-0.7,0c-0.1,0-0.2,0-0.2,0c-0.1,0.1,0,0.1,0,0.2c0,0.3,0,0.5,0,0.8c0,0.1,0,0.1-0.1,0.1H9.2c-0.3,0-0.5,0-0.8,0
		c-0.1,0-0.1,0-0.1,0.1c0,0.2,0,0.4,0,0.7c0,0.1,0,0.1,0,0.2s0,0.1-0.1,0.1H8.1c-0.3,0-0.5,0-0.8,0C7,5.9,7,5.9,7,6
		c0,0.3,0,0.6,0,0.8v0.1C7,7,6.9,7,6.9,7C6.8,7,6.8,7,6.7,7H0.2C0,7,0,7,0,7.3c0,3.3,0,6.6,0,9.9c0,0.2,0,0.2,0.2,0.2h6.5h0.1
		c0.1,0,0.1,0,0.1,0.1v0.1c0,0.3,0,0.5,0,0.8c0,0.2,0,0.2,0.2,0.2s0.5,0,0.7,0c0.3,0,0.2,0,0.2,0.2c0,0.3,0,0.5,0,0.8
		c0,0.1,0,0.2,0.2,0.2s0.3,0,0.5,0c0.1,0,0.3,0,0.4,0c0.1,0,0.1,0,0.1,0.1V20c0,0.3,0,0.5,0,0.8c0,0.1,0,0.1,0.1,0.1
		c0.3,0,0.5,0,0.8,0h0.1c0,0,0.1,0,0.1,0.1v0.1c0,0.3,0,0.6,0,0.8c0,0.1,0,0.1,0.1,0.1c0.3,0,0.6,0,0.9,0c0,0,0,0,0.1,0
		c0,0,0.1,0,0.1-0.1s0-0.1,0-0.2v-19C11.6,2.7,11.6,2.6,11.6,2.5z M10.4,6.9c0,0.1,0,0.1-0.1,0.1C10,7,9.8,7,9.5,7S9.3,7,9.3,7.3
		s0,0.5,0,0.8c0,0.2,0,0.2-0.2,0.2h-1c0,0.3,0,0.6,0,0.9c0,0.1,0,0.1,0,0.2c0,0,0,0.1-0.1,0.1l0,0H7.9H3.7c-0.2,0-0.2,0-0.2-0.2
		c0-0.3,0-0.5,0-0.8V8.4c0,0,0-0.1,0.1-0.1h0.1c1.4,0,2.9,0,4.3,0h0.1v-1V7.2c0-0.1,0-0.1,0.1-0.1h0.1c0.3,0,0.5,0,0.8,0
		c0.2,0,0.2,0,0.2-0.2c0-0.3,0-0.5,0-0.8V6c0-0.1,0-0.1,0.1-0.1c0.3,0,0.6,0,0.9,0c0.1,0,0.1,0,0.1,0.1C10.4,6.3,10.4,6.6,10.4,6.9z
		"/>
	<path d="M17.5,14.3c-0.3,0-0.7,0-1,0c-0.1,0-0.1,0-0.1,0.1c0,0.2,0,0.3,0,0.5l0,0c0,0.1,0,0.3,0,0.4c0,0.2,0,0.2-0.2,0.2
		c-0.3,0-0.6,0-0.8,0c-0.1,0-0.1,0-0.1,0.1c0,0.3,0,0.6,0,0.9c0,0,0,0,0,0.1s0,0.1,0.1,0.1h0.1h0.9h0.1c0,0,0.1,0,0.1-0.1l0,0v-0.1
		c0-0.3,0-0.5,0-0.8s0-0.2,0.2-0.2c0.3,0,0.6,0,0.8,0c0.1,0,0.1,0,0.1-0.1c0-0.3,0-0.7,0-1C17.6,14.3,17.6,14.3,17.5,14.3z"/>
	<path d="M20.1,11.7C20.1,11.8,20.1,11.8,20.1,11.7c0.1,0.1,0.1,0.1,0.2,0.1c0.3,0,0.6,0,0.9,0h0.1c0,0,0.1,0,0.1-0.1l0,0v-0.1
		c0-0.3,0-0.5,0-0.8s0-0.2,0.2-0.2c0.3,0,0.6,0,0.8,0c0.1,0,0.1,0,0.1-0.1c0-0.3,0-0.7,0-1c0-0.1,0-0.1-0.1-0.1c-0.3,0-0.7,0-1,0
		c-0.1,0-0.1,0-0.1,0.1c0,0.2,0,0.3,0,0.5l0,0c0,0.1,0,0.3,0,0.4c0,0.2,0,0.2-0.2,0.2c-0.3,0-0.6,0-0.8,0c-0.1,0-0.1,0-0.1,0.1
		C20.1,11,20.1,11.4,20.1,11.7L20.1,11.7z"/>
	<path d="M22.7,9.4c0.3,0,0.7,0,1,0c0.1,0,0.1,0,0.1-0.1c0-0.3,0-0.7,0-1c0-0.1,0-0.1-0.1-0.1c-0.2,0-0.3,0-0.5,0s-0.3,0-0.5,0
		c-0.1,0-0.1,0-0.1,0.1c0,0.3,0,0.7,0,1C22.6,9.4,22.6,9.4,22.7,9.4z"/>
	<path d="M24,16.3c0-0.3,0-0.6,0-0.9v-0.1c0,0,0-0.1-0.1-0.1l0,0h-0.1c-0.3,0-0.5,0-0.8,0s-0.2,0-0.2-0.2c0-0.3,0-0.6,0-0.8
		c0-0.1,0-0.1-0.1-0.1c-0.3,0-0.7,0-1,0.1c-0.1,0-0.1,0-0.1,0.1c0,0.3,0,0.7,0.1,1c0,0.1,0,0.1,0.1,0.1c0.2,0,0.3,0,0.5,0l0,0
		c0.1,0,0.3,0,0.4,0c0.2,0,0.2,0,0.2,0.2c0,0.3,0,0.6,0,0.8c0,0.1,0,0.1,0.1,0.1c0.3,0,0.6,0,0.9,0c0,0,0,0,0.1,0
		C24,16.5,24,16.4,24,16.3C24,16.4,24,16.3,24,16.3z"/>
	<path d="M21.4,14c0-0.3,0-0.6,0-0.9V13c0,0,0-0.1-0.1-0.1l0,0h-0.1c-0.3,0-0.5,0-0.8,0s-0.2,0-0.2-0.2c0,0,0,0,0-0.1
		c0-0.3,0-0.5,0-0.8c0-0.1,0-0.1-0.1-0.1c0,0,0,0-0.1,0h-0.1c-0.3,0-0.6,0-0.8,0c-0.1,0-0.1,0-0.1,0.1c-0.1,0-0.1,0-0.1,0.1
		c0,0.3,0,0.5,0,0.8v0.1c0,0.2,0,0.2-0.2,0.2c-0.3,0-0.6,0-0.8,0c-0.1,0-0.1,0-0.1,0.1c0,0.3,0,0.6,0,0.9c0,0,0,0,0,0.1
		s0,0.1,0.1,0.1H18c0.3,0,0.6,0,0.9,0H19c0,0,0.1,0,0.1-0.1l0,0V14c0-0.3,0-0.5,0-0.8c0-0.1,0-0.1,0-0.2c0,0,0,0,0.1,0
		c0.2,0,0.3,0,0.5,0l0,0H20h0h0.1c0.1,0,0.1,0,0.1,0.2c0,0.3,0,0.6,0,0.8c0,0.1,0,0.1,0.1,0.1c0.3,0,0.6,0,0.9,0c0,0,0,0,0.1,0
		C21.4,14.2,21.4,14.1,21.4,14C21.4,14.1,21.4,14,21.4,14z"/>
	<path d="M18.7,11.9L18.7,11.9c0.1,0,0.1,0,0.1-0.1v-0.1c0-0.3,0-0.6,0-0.9v-0.1c0,0,0-0.1-0.1-0.1l0,0h-0.1c-0.3,0-0.5,0-0.8,0
		s-0.2,0-0.2-0.2c0-0.3,0-0.6,0-0.8c0-0.1,0-0.1-0.1-0.1c-0.3,0-0.7,0-1,0.1c-0.1,0-0.1,0-0.1,0.1c0,0.3,0,0.7,0.1,1
		c0,0.1,0,0.1,0.1,0.1c0.2,0,0.3,0,0.5,0l0,0c0.1,0,0.3,0,0.4,0c0.2,0,0.2,0,0.2,0.2c0,0.3,0,0.6,0,0.8c0,0.1,0,0.1,0.1,0.1
		C18,11.9,18.4,11.9,18.7,11.9z"/>
	<path d="M16.2,9.5c0-0.3,0-0.7-0.1-1c0-0.1,0-0.1-0.1-0.1c-0.3,0-0.7,0-1,0.1c-0.1,0-0.1,0-0.1,0.1c0,0.2,0,0.3,0,0.5s0,0.3,0,0.5
		c0,0.1,0,0.1,0.1,0.1c0.3,0,0.7,0,1-0.1C16.2,9.6,16.2,9.6,16.2,9.5z"/>
</g>
</svg>
