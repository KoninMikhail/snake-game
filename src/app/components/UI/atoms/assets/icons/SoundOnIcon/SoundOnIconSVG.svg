<?xml version="1.0" encoding="utf-8"?>
<!-- Generator: Adobe Illustrator 26.0.1, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
<svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
	 viewBox="0 0 24 24" style="enable-background:new 0 0 24 24;" xml:space="preserve">
<g id="Layer_1_00000013189198407605598220000002630086399544872380_">
	<path d="M11.6,2.5c0-0.1,0-0.1-0.1-0.1c-0.3,0-0.6,0-0.9,0c-0.1,0-0.1,0-0.1,0.1v0.1c0,0.3,0,0.5,0,0.8v0.1c0,0,0,0.1-0.1,0.1l0,0
		h-0.1c-0.2,0-0.5,0-0.7,0c-0.1,0-0.2,0-0.2,0c-0.1,0.1,0,0.1,0,0.2c0,0.3,0,0.5,0,0.8c0,0.1,0,0.1-0.1,0.1H9.2c-0.3,0-0.5,0-0.8,0
		c-0.1,0-0.1,0-0.1,0.1c0,0.2,0,0.4,0,0.7c0,0.1,0,0.1,0,0.2s0,0.1-0.1,0.1H8.1c-0.3,0-0.5,0-0.8,0C7,5.9,7,5.9,7,6
		c0,0.3,0,0.6,0,0.8v0.1C7,7,6.9,7,6.9,7C6.8,7,6.8,7,6.7,7H0.2C0,7,0,7,0,7.3c0,3.3,0,6.6,0,9.9c0,0.2,0,0.2,0.2,0.2h6.5h0.1
		c0.1,0,0.1,0,0.1,0.1v0.1c0,0.3,0,0.5,0,0.8c0,0.2,0,0.2,0.2,0.2s0.5,0,0.7,0c0.3,0,0.2,0,0.2,0.2c0,0.3,0,0.5,0,0.8
		c0,0.1,0,0.2,0.2,0.2s0.3,0,0.5,0c0.1,0,0.3,0,0.4,0c0.1,0,0.1,0,0.1,0.1V20c0,0.3,0,0.5,0,0.8c0,0.1,0,0.1,0.1,0.1
		c0.3,0,0.5,0,0.8,0h0.1c0,0,0.1,0,0.1,0.1v0.1c0,0.3,0,0.6,0,0.8c0,0.1,0,0.1,0.1,0.1c0.3,0,0.6,0,0.9,0c0,0,0,0,0.1,0
		c0,0,0.1,0,0.1-0.1s0-0.1,0-0.2v-19C11.6,2.7,11.6,2.6,11.6,2.5z M10.4,6.9c0,0.1,0,0.1-0.1,0.1C10,7,9.8,7,9.5,7S9.3,7,9.3,7.3
		s0,0.5,0,0.8c0,0.2,0,0.2-0.2,0.2h-1c0,0.3,0,0.6,0,0.9c0,0.1,0,0.1,0,0.2c0,0,0,0.1-0.1,0.1l0,0H7.9H3.7c-0.2,0-0.2,0-0.2-0.2
		c0-0.3,0-0.5,0-0.8V8.4c0,0,0-0.1,0.1-0.1h0.1c1.4,0,2.9,0,4.3,0h0.1v-1V7.2c0-0.1,0-0.1,0.1-0.1h0.1c0.3,0,0.5,0,0.8,0
		c0.2,0,0.2,0,0.2-0.2c0-0.3,0-0.5,0-0.8V6c0-0.1,0-0.1,0.1-0.1c0.3,0,0.6,0,0.9,0c0.1,0,0.1,0,0.1,0.1C10.4,6.3,10.4,6.6,10.4,6.9z
		"/>
	<path d="M22.7,9.4c0.3,0,0.7,0,1,0c0.1,0,0.1,0,0.1-0.1c0-0.3,0-0.7,0-1c0-0.1,0-0.1-0.1-0.1c-0.2,0-0.3,0-0.5,0s-0.3,0-0.5,0
		c-0.1,0-0.1,0-0.1,0.1c0,0.3,0,0.7,0,1C22.6,9.4,22.6,9.4,22.7,9.4z"/>
	<path d="M16.2,9.5c0-0.3,0-0.7-0.1-1c0-0.1,0-0.1-0.1-0.1c-0.3,0-0.7,0-1,0.1c-0.1,0-0.1,0-0.1,0.1c0,0.2,0,0.3,0,0.5s0,0.3,0,0.5
		c0,0.1,0,0.1,0.1,0.1c0.3,0,0.7,0,1-0.1C16.2,9.6,16.2,9.6,16.2,9.5z"/>
</g>
<g id="Layer_2_00000179614502495302775930000017321550740226036372_">
	<path d="M22.7,16c0-0.1,0-0.1,0-0.2c0-2.5,0-5,0-7.5c0-0.2,0-0.2,0.2-0.2c0.3,0,0.5,0,0.8,0c0.1,0,0.1,0,0.1,0.1v0.1
		c0,2.5,0,4.9,0,7.4c0,0.3,0,0.2-0.3,0.3C23.3,16,23,16,22.7,16L22.7,16z"/>
	<path d="M20.2,12c0,1.2,0,2.5,0,3.7c0,0.3,0,0.2-0.2,0.2c-0.3,0-0.5,0-0.8,0C19,16,19,16,19,15.8v-0.1c0-2.5,0-5,0-7.5
		C19,8,19,8,19.3,8c0.2,0,0.5,0,0.7,0c0.1,0,0.1,0,0.1,0.1v0.1L20.2,12z"/>
	<path d="M15.3,12c0-0.9,0-1.8,0-2.6c0-0.2,0-0.2-0.2-0.2s-0.5,0-0.7,0c-0.1,0-0.1,0-0.1-0.1c0-0.3,0-0.6,0-0.9c0-0.1,0-0.1,0.1-0.1
		c0.3,0,0.6,0,0.9,0c0.1,0,0.1,0,0.1,0.1c0,0.3,0,0.5,0,0.8c0,0,0,0,0,0.1s0,0.1,0.1,0.1c0.2,0,0.5,0,0.7,0h0.1c0.1,0,0.1,0,0.1,0.1
		v0.1c0,1.8,0,3.5,0,5.3c0,0.2,0,0.2-0.2,0.2s-0.5,0-0.7,0c-0.1,0-0.1,0-0.1,0.1c0,0.3,0,0.6,0,0.8v0.1c0,0,0,0.1-0.1,0.1l0,0
		c-0.3,0-0.6,0-0.9,0c-0.1,0-0.1,0-0.1-0.1c0-0.3,0-0.6,0-0.9c0-0.1,0-0.1,0.1-0.1c0.2,0,0.5,0,0.7,0h0.1c0.1,0,0.1,0,0.1-0.1v-0.1
		L15.3,12z"/>
	<path d="M22.7,16c0,0.1,0,0.1,0,0.2c0,0.6,0,1.3,0,1.9c0,0.2,0,0.2-0.2,0.2c-0.3,0-0.5,0-0.8,0h-0.1c0,0-0.1,0-0.1-0.1v-0.1v-1.9
		c0-0.1,0-0.1,0.2-0.1c0.3,0,0.5,0,0.8,0C22.6,16,22.6,16,22.7,16L22.7,16z"/>
	<path d="M17.9,16.5c0-0.2,0-0.3,0-0.5c0-0.1,0-0.1,0.1-0.1c0.3,0,0.6,0,0.9,0c0.1,0,0.1,0,0.1,0.1c0,0.3,0,0.6,0,0.9
		c0,0.1,0,0.1-0.1,0.1c-0.3,0-0.5,0-0.8,0c-0.2,0-0.2,0-0.2,0.2s0,0.5,0,0.7V18c0,0,0,0.1-0.1,0.1l0,0h-0.1H17c0,0,0,0-0.1,0
		c0,0-0.1,0-0.1-0.1c0,0,0,0,0-0.1c0-0.3,0-0.6,0-0.9c0-0.1,0-0.1,0.1-0.1c0.3,0,0.5,0,0.8,0c0.2,0,0.2,0,0.2-0.2
		C17.9,16.8,17.9,16.7,17.9,16.5L17.9,16.5z"/>
	<path d="M21.6,6.9V6V5.9c0,0,0-0.1,0.1-0.1s0.2,0,0.4,0c0.2,0,0.4,0,0.6,0c0.1,0,0.1,0,0.1,0.1s0,0.2,0,0.2c0,0.6,0,1.1,0,1.7v0.1
		c0,0.1,0,0.1-0.1,0.1c-0.2,0-0.4,0-0.5,0S22,8,21.8,8c-0.1,0-0.1,0-0.1-0.1c0-0.3,0-0.6,0-0.9L21.6,6.9z"/>
	<path d="M21,4.6c0.1,0,0.3,0,0.4,0s0.1,0,0.1,0.1c0,0.3,0,0.6,0,0.9c0,0.1,0,0.1-0.1,0.1c-0.3,0-0.6,0-0.9,0c-0.1,0-0.1,0-0.1-0.1
		c0-0.3,0-0.6,0-0.9c0-0.1,0-0.1,0.1-0.1C20.7,4.6,20.8,4.6,21,4.6z"/>
	<path d="M17.9,7.5c0-0.1,0-0.3,0-0.4c0-0.2,0-0.2,0.2-0.2s0.5,0,0.7,0c0.1,0,0.1,0,0.2,0c0,0,0,0,0,0.1c0,0.2,0,0.4,0,0.6
		c0,0.1,0,0.2,0,0.2C19,8,19,8,18.9,8c-0.2,0-0.5,0-0.7,0c-0.1,0-0.1,0-0.2,0s-0.1,0-0.1-0.1C17.9,7.8,17.9,7.6,17.9,7.5L17.9,7.5
		L17.9,7.5z"/>
	<path d="M21,18.2c0.2,0,0.3,0,0.5,0c0.1,0,0.1,0,0.1,0.1c0,0.3,0,0.6,0,0.9c0,0.1,0,0.1-0.1,0.1c-0.3,0-0.6,0-0.9,0
		c-0.1,0-0.1,0-0.1-0.1c0-0.3,0-0.6,0-0.9c0-0.1,0-0.1,0.1-0.1S20.8,18.2,21,18.2z"/>
	<path d="M17.3,6.9c-0.1,0-0.3,0-0.4,0s-0.1,0-0.1-0.1c0-0.3,0-0.6,0-0.9c0-0.1,0-0.1,0.1-0.1c0.3,0,0.6,0,0.9,0
		c0.1,0,0.1,0,0.1,0.1c0,0.3,0,0.6,0,0.9c0,0.1,0,0.1-0.1,0.1C17.6,6.9,17.5,6.9,17.3,6.9z"/>
</g>
</svg>
