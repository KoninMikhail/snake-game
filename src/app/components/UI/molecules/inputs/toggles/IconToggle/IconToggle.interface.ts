import React, {ReactElement} from 'react';
import {IconSize} from '@ui/atoms/assets/icons/Icon.types';

// ISoundControlProps
// ----------------------------------------------------------------------
export interface IIconToggle {
    state: boolean;
    iconWhenTrue: ReactElement;
    iconWhenFalse: ReactElement;
    size: IconSize;
    onClick?: (e: React.MouseEvent) => void;
}
