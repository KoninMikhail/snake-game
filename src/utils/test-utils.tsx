import React, {PropsWithChildren} from 'react';
import type {RenderOptions} from '@testing-library/react';
import {render} from '@testing-library/react';
import type {PreloadedState} from '@reduxjs/toolkit';
import {configureStore} from '@reduxjs/toolkit';
import {Provider} from 'react-redux';

import type {AppStore, RootState} from '@store/store';
import appSlice from '@store/slices/app.slice';
import soundSlice from '@store/slices/sound.slice';
import maxScoreSlice from '@store/slices/maxScore.slice';
import gameDataSlice from '@store/slices/game.slice';
import {ImagesData} from '@data/imagesData';
import {mockImagesData, mockSoundData} from '@/setupTests';
import SoundsProvider from '@context/SoundsContext';
import ImagesProvider from '@context/ImagesContext';
import {SoundsData} from '@data/soundsData';
import {DefaultTheme, ThemeProvider} from 'styled-components';
import {theme as appTheme} from '@styles/theme';

// As a basic setup, import your same slice reducers

// This type interface extends the default options for render from RTL, as well
// as allows the user to specify other things such as initialState, store.
interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
    preloadedState?: PreloadedState<RootState>;
    store?: AppStore;
    images?: ImagesData;
    sounds?: SoundsData;
    theme?: DefaultTheme;
}

export function renderWithProviders(
    ui: React.ReactElement,
    {
        preloadedState = {},
        // Automatically create a store instance if no store was passed in
        store = configureStore({
            reducer: {
                app: appSlice,
                sounds: soundSlice,
                maxScore: maxScoreSlice,
                game: gameDataSlice,
            },
            preloadedState,
        }),
        images = mockImagesData,
        sounds = mockSoundData,
        theme = appTheme,
        ...renderOptions
    }: ExtendedRenderOptions = {}
) {
    function Wrapper({children}: PropsWithChildren): JSX.Element {
        return (
            <Provider store={store}>
                <ThemeProvider theme={theme}>
                    <SoundsProvider sounds={sounds}>
                        <ImagesProvider images={images}>{children}</ImagesProvider>
                    </SoundsProvider>
                </ThemeProvider>
            </Provider>
        );
    }

    // Return an object with the store and all of RTL's query functions
    return {store, ...render(ui, {wrapper: Wrapper, ...renderOptions})};
}
