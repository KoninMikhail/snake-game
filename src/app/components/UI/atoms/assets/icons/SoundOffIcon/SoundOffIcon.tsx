import { ReactComponent as SoundOffIconSVG } from './SoundOffIconSVG.svg';
import { useTheme } from 'styled-components';
import getThemeIconSize from '@/helpers/theme/getThemeIconSize';
import { IIcon } from '@ui/atoms/assets/icons/Icon.interface';

/**
 * @name SoundOffIconSVG
 * @author Mikhail Konin <dev.konin@gmail.com>
 *
 * Use as icon.
 *
 * @param size - icon size
 * @param rest - optional arguments for passing to icon
 * @returns {JSX.Element}
 */
export const SoundOffIcon = ({ size = 'md', ...rest }: IIcon) => {
    const theme = useTheme();
    const outputSize = getThemeIconSize(theme, size);
    return (
        <>
            <SoundOffIconSVG
                fill={'currentColor'}
                width={outputSize}
                height={outputSize}
                {...rest}
            />
        </>
    );
};
