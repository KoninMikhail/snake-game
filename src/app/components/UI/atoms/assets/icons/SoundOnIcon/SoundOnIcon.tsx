import { ReactComponent as SoundOnIconSVG } from './SoundOnIconSVG.svg';
import { useTheme } from 'styled-components';
import getThemeIconSize from '@/helpers/theme/getThemeIconSize';
import { IIcon } from '@ui/atoms/assets/icons/Icon.interface';

/**
 * @name SoundOnIcon
 * @author Mikhail Konin <dev.konin@gmail.com>
 *
 * Use as icon.
 *
 * @param size - icon size
 * @param rest - optional arguments for passing to icon
 */

export const SoundOnIcon = ({ size = 'md', ...rest }: IIcon) => {
    const theme = useTheme();
    const outputSize = getThemeIconSize(theme, size);
    return (
        <>
            <SoundOnIconSVG
                fill={'currentColor'}
                width={outputSize}
                height={outputSize}
                {...rest}
            />
        </>
    );
};
