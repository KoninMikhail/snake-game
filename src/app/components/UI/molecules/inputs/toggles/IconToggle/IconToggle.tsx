import {
    IconToggleContainer,
    IconToggleIconHolder,
} from '@ui/molecules/inputs/toggles/IconToggle/style/StyledIconToggle';
import { IIconToggle } from '@ui/molecules/inputs/toggles/IconToggle/IconToggle.interface';
import { useTheme } from 'styled-components';
import { cloneElement } from 'react';
import getThemeIconSize from '@helpers/theme/getThemeIconSize';

export const IconToggle = ({ size, iconWhenTrue, iconWhenFalse, state, onClick }: IIconToggle) => {
    const theme = useTheme();
    const themeIconSizeValue = getThemeIconSize(theme, size);
    return (
        <IconToggleContainer size={themeIconSizeValue} onClick={onClick}>
            <IconToggleIconHolder visible={state}>
                {cloneElement(iconWhenTrue, { size: size })}
            </IconToggleIconHolder>
            <IconToggleIconHolder visible={!state}>
                {cloneElement(iconWhenFalse, { size: size })}
            </IconToggleIconHolder>
        </IconToggleContainer>
    );
};
