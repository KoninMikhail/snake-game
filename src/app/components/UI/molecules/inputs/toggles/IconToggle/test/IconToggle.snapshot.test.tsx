import { renderWithProviders } from '@utils/test-utils';
import { IconToggle } from '@ui/molecules/inputs/toggles/IconToggle/IconToggle';
import { SoundOnIcon } from '@ui/atoms/assets/icons/SoundOnIcon/SoundOnIcon';
import { SoundOffIcon } from '@ui/atoms/assets/icons/SoundOffIcon/SoundOffIcon';

jest.mock('@ui/atoms/assets/icons/SoundOnIcon/SoundOnIcon', () => {
    return {
        SoundOnIcon: () => <>{'soundOnIcon'}</>,
    };
});
jest.mock('@ui/atoms/assets/icons/SoundOffIcon/SoundOffIcon', () => {
    return {
        SoundOffIcon: () => <>{'soundOffIcon'}</>,
    };
});

/**
 * Tests
 */
describe('IconToggle', () => {
    it('equal snapshot when true', () => {
        const { container } = renderWithProviders(
            <IconToggle
                iconWhenFalse={<SoundOffIcon />}
                iconWhenTrue={<SoundOnIcon />}
                size={'md'}
                state
            />
        );
        expect(container).toMatchSnapshot();
    });

    it('equal snapshot when false', () => {
        const { container } = renderWithProviders(
            <IconToggle
                iconWhenFalse={<SoundOffIcon />}
                iconWhenTrue={<SoundOnIcon />}
                size={'md'}
                state={false}
            />
        );
        expect(container).toMatchSnapshot();
    });
});
