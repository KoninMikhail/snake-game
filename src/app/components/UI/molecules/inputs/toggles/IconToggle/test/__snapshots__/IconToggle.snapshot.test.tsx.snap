// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconToggle equal snapshot when false 1`] = `
.c0 {
  cursor: pointer;
  display: block;
  position: relative;
  width: 1em;
  height: 1em;
}

.c1 {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  width: 100%;
  visibility: hidden;
  opacity: 0;
}

.c2 {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  width: 100%;
  visibility: visible;
  opacity: 1;
}

<div>
  <div
    class="c0"
  >
    <span
      class="c1"
    >
      soundOnIcon
    </span>
    <span
      class="c2"
    >
      soundOffIcon
    </span>
  </div>
</div>
`;

exports[`IconToggle equal snapshot when true 1`] = `
.c0 {
  cursor: pointer;
  display: block;
  position: relative;
  width: 1em;
  height: 1em;
}

.c1 {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  width: 100%;
  visibility: visible;
  opacity: 1;
}

.c2 {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  width: 100%;
  visibility: hidden;
  opacity: 0;
}

<div>
  <div
    class="c0"
  >
    <span
      class="c1"
    >
      soundOnIcon
    </span>
    <span
      class="c2"
    >
      soundOffIcon
    </span>
  </div>
</div>
`;
